src/evm-block-extractor/src/task/block_extractor.rs

```diff
@@ -221,4 +243,123 @@ impl<C: Client> BlockExtractor<C> {
 
         Ok(())
     }
+
+    /// This function:
+    /// - checks if the `new_blocks` sequence have correct hashes.
+    /// - checks if `last_stored_block.hash == new_blocks[0].prev_block_hash`.
+    async fn validate_chain(
```

Comment:
this function can be simplified. You only have to loop the new blocks backwards and check the parent against the previous block. Once you reach the first block in the array, compare it with the latest block in the database. In addition, if you pass the latest block in the database as an argument, you can (and should) unit-test this logic properly.

Reply:
Done, using iterators for simplification